src/lib.rs

```diff
@@ -175,6 +177,17 @@ pub struct ColumnStack<T: Columnation> {
 }
 
 impl<T: Columnation> ColumnStack<T> {
+    /// Construct a [ColumnStack], reserving space for `capacity` elements
+    ///
+    /// Note that the associated region is not initialized to a specific capacity
+    /// because we can't generally know how much space would be required.
+    pub fn with_capacity(capacity: usize) -> Self {
+        Self {
+            local: Vec::with_capacity(capacity),
+            inner: T::InnerRegion::default(),
+        }
+    }
+
```

Comment:
I think we discussed this, and IIRC one take was that regions could have a `fn with_capacity_for(&[T]) -> Self` or something analogous, maybe an iterator, which would enable a better `clone_from` implementation. If we concluded that, no need to fix here but it might be worth leaving a note (IIRC, this method mainly exists to be used by another PR elsewhere, I think?).

Reply:
The function is private now, because it's useful internally but doesn't do what a caller would expect it to do. A `fn with_capacity_for(&[T]) -> Self` seems useful, but it would require support of the regions, which we currently don't have. I'll leave this for later.